Rfc

kajima-genius opened the pull request.

kajima-genius committed:
the normal version, which is needed due to errors in the commits

kajima-genius committed:
code formatting

dmkofgl commented on src/main/java/com/example/music/backend/user/controller/UserController.java:
```diff
@@ -5,64 +5,58 @@
-    public String getHomePage() {
-        return "home";
-    }
+//    @GetMapping(path = "/user/registration")
+//    public String getPage(Model model) {
+//        model.addAttribute("UserDto", new UserDto());
+//        return "registration";
+//    }
+//
+//    @GetMapping(path = "/login")
+//    public String getLoginPage() {
+//        return "login";
+//    }
+//
+//    @GetMapping(path = "/home")
+//    public String getHomePage() {
+//        return "home";
+//    }
```
Why do you need this comment?

kajima-genius replied:
deleted

dmkofgl commented on src/main/java/com/example/music/backend/user/controller/UserController.java:
```diff
@@ -5,64 +5,58 @@
             User saved = service.create(userDto);
             String appUrl = request.getContextPath();
             eventPublisher.publishEvent(new OnRegistrationCompleteEvent(saved, request.getLocale(), appUrl));
+            status = HttpStatus.OK;
```
Repeat status codes, please. There is one more suitable for this case.

dmkofgl commented on src/main/java/com/example/music/backend/user/controller/UserController.java:
```diff
@@ -5,64 +5,58 @@
+            status = HttpStatus.BAD_REQUEST;
         }
-        return mav;
+        return ResponseEntity.status(status).build();
```
It would be better to have a response body if it has an error then show some error message, if it finished successfully - show what had been created

kajima-genius replied:
now returning ResponseEntity.created(URI.create("/users" + saved.getId())).build()

dmkofgl commented on src/main/java/com/example/music/backend/user/controller/UserController.java:
```diff
@@ -5,64 +5,58 @@
         } catch (UserAlreadyExistException uaeEx) {
-            mav.setViewName("registration");
-            mav.addObject("message", "An account for that username/email already exists.");
+            status = HttpStatus.BAD_REQUEST;
```
This catch is the same as for RuntimeException, do you really need it?

kajima-genius replied:
now the service is responsible for this

dmkofgl commented on src/main/java/com/example/music/backend/user/controller/UserController.java:
```diff
@@ -5,64 +5,58 @@
+                                             HttpServletRequest request) {
+        HttpStatus status;
         try {
             String encodedPassword = encoder.encode(userDto.getPassword());
             userDto.setPassword(encodedPassword);
```
What about using Service class to update a User object?

kajima-genius committed:
changed UserController:
1) Changed return
2) Error handling is now in the service

kajima-genius committed:
the following changes:
1) In UserController, in the create method, the created object is returned
2) Added id to UserDto constructor
3) Added getDto method to UserService
4) Moved password encoding logic to create method in UserServiceImpl

dmkofgl commented on src/main/java/com/example/music/backend/user/service/UserServiceImpl.java:
```diff
@@ -44,4 +51,11 @@ public void processOAuthPostLogin(OAuth2User oAuth2User) {
     }
+
+    @Override
+    public UserDto getDto(String email) {
```
Keep union strategy. Service should return entity or DTO/Response classes. Don't mix it, please

dmkofgl commented on src/main/java/com/example/music/backend/user/dto/UserDto.java:
```diff
@@ -21,11 +20,12 @@
         this.email = email;
         this.password = password;
         this.dateOfBirth = dateOfBirch;
+        this.id = id;
```
Why do you need it inside DTO?

kajima-genius replied:
I agree here id is not needed, I am doing PlaylistDto and id is needed there, confused
